src/compiler/sys.ts

```diff
@@ -569,7 +573,31 @@ namespace ts {
                     }
                 },
                 setTimeout,
-                clearTimeout
+                clearTimeout,
+                require(initialDir: string, moduleName: string) {

```

Comment:
can we move `require` from `sys` to `ServerHost`?
-it looks like currently only tsserver will use it
- if we do it in tsserver we can use our module resolution implementation and then do `require` with resolved path without relying on undocumented properties